packages/slate/src/models/node.js

```diff
@@ -1976,6 +1976,25 @@ class Node {
   validate(schema) {
     return schema.validateNode(this)
   }
+
+  /**
+   * The first descendant key requiring validation
+   *
+   * @param {Schema} schema
+   * @return {String|Null}
+   */
+
+  getFirstInvalidDescendantKey(schema) {
+    if (this.validate(schema)) {
+      return this.key
+    }
+    let result = null
+    this.nodes.find(n => {
+      result = n.getFirstInvalidDescendantKey(schema)
+      return typeof result === 'string'
```

Comment:
This check is confusing. Since we know this is either going to return a node or `null`, we should explicitly do `return !!result` to be less strange. Or actually, it could all be simplified to:

```js
this.nodes.find(n => n.getFirstInvalidDescendant(schema))
```